models/aeif_cond_beta_multisynapse.h

```diff
@@ -208,14 +209,62 @@ class aeif_cond_beta_multisynapse : public Archiving_Node
   void get_status( DictionaryDatum& ) const;
   void set_status( const DictionaryDatum& );
 
+  static inline Name
+  get_g_receptor_name( size_t receptor )
+  {
+    std::stringstream receptor_name;
+    receptor_name << "g_" << receptor + 1;
+    return Name( receptor_name.str() );
+  };
+
+  //! Class that reads out state vector elements, used by UniversalDataLogger
+  class DataAccessFunctor
```

Comment:
Does this class need to be public? I am in doubt if it is sensible to declare this class here, useful only for `aeif_cond_beta_multisynapse`. Could one use a template function instead? That would avoid writing a lot of boilerplate code in models.

Reply:
You're right. `DataAccessFunctor` should be a class template. I moved its implementation to `recordables_map.h`, because it is used by the `DynamicRecordablesMap` class template.
I also had to change the way in which `DataAccessFunctor` instances were created by the `aeif_cond_beta_multisynapse`, and also how the `DataAccessFunctor` queried the private state of the `HostNode`. Now `aeif_cond_beta_multisynapse` has two new functions `get_data_access_functor` and `get_state_element`. The latter returns the value of the recorded state element. The former creates a `DataAccessFunctor<aeif_cond_beta_multisynapse>`, that is intended as the supplier for the `DynamicRecordablesMap`.